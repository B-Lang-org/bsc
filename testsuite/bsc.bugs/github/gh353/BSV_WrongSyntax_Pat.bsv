typedef struct {
  Bool f1;
  Bool f2;
} S;

typedef union tagged {
  struct { Bool f1; Bool f2; } C;
} U;

// -----
// Try using struct syntax for a constructor

function Bool fn1 (U v);
  case (v) matches
    C { f1: .x, f2: .y }: return (x && y);
    default: return False;
  endcase
endfunction

// -----
// Try using constructor syntax for a struct

function Bool fn2 (S v);
  case (v) matches
    tagged S { f1: .x, f2: .y }: return (x && y);
    default: return False;
  endcase
endfunction
