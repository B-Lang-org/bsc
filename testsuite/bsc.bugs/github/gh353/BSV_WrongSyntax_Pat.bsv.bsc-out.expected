checking package dependencies
compiling BSV_WrongSyntax_Pat.bsv
Error: "BSV_WrongSyntax_Pat.bsv", line 15, column 5: (T0007)
  Unbound type constructor `C'
Error: "BSV_WrongSyntax_Pat.bsv", line 25, column 12: (T0003)
  Unbound constructor `S'
